libs/wingsdk/src/http/http.ts

```diff
@@ -0,0 +1,259 @@
+   */
+  readonly body: string;
+}
+/**
+ * default options to attach to any request
+ */
+const defaultOptions: RequestOptions = {
+  method: HttpMethods.GET,
+  headers: {},
+  cache: RequestCache.DEFAULT,
+  redirect: RequestRedirect.FOLLOW,
+};
+/**
+ * the Http class is used for calling different HTTP methods and requesting and sending information online,
+ *  as well as testing public accessible resources
+ * @inflight
+ */
+export class Util {
+  /**
+   * @internal
+   */
+  public static _toInflightType(): Code {
+    return InflightClient.forType(__filename, this.name);
+  }
+  /**
+   * Sends an http request of any method.
+   * @param url
+   * @param options
+   * @inflight
+   * @returns the formatted response of the call
+   */
+  static async fetch(url: string, options?: RequestOptions): Promise<Response> {
+    const res = await fetch(url, { ...defaultOptions, ...options });
+    return this._formatResponse(res);
+  }
+  /**
+   * Sends a GET request.
+   * @param url
+   * @param options
+   * @inflight
+   * @returns the formatted response of the call
+   */
+  static async get(url: string, options?: RequestOptions): Promise<Response> {
+    return this.fetch(url, {
+      ...defaultOptions,
+      ...options,
+      method: HttpMethods.GET,
+    });
+  }
+  /**
+   * Sends a POST request.
+   * @param url
+   * @param options
+   * @inflight
+   * @returns the formatted response of the call
+   */
+  static async post(url: string, options?: RequestOptions): Promise<Response> {
+    return this.fetch(url, {
+      ...defaultOptions,
+      ...options,
+      method: HttpMethods.POST,
+    });
+  }
+  /**
+   * Sends a PUT request.
+   * @param url
+   * @param options
+   * @inflight
+   * @returns the formatted response of the call
+   */
+  static async put(url: string, options?: RequestOptions): Promise<Response> {
+    return this.fetch(url, {
+      ...defaultOptions,
+      ...options,
+      method: HttpMethods.PUT,
+    });
+  }
+  /**
+   * Sends a PATCH request.
+   * @param url
+   * @param options
+   * @inflight
+   * @returns the formatted response of the call
+   */
+  static async patch(url: string, options?: RequestOptions): Promise<Response> {
+    return this.fetch(url, {
+      ...defaultOptions,
+      ...options,
+      method: HttpMethods.PATCH,
+    });
+  }
+
+  /**
+   * Sends a DELETE request.
+   * @param url
+   * @param options
+   * @inflight
+   * @returns the formatted response of the call
+   */
+  static async delete(
+    url: string,
+    options?: RequestOptions
+  ): Promise<Response> {
+    return this.fetch(url, {
+      ...defaultOptions,
+      ...options,
+      method: HttpMethods.DELETE,
+    });
+  }
+
+  private static async _formatResponse(
+    response: globalThis.Response
+  ): Promise<Response> {
+    const headers: Record<string, string> = {};
+    response.headers?.forEach((val: string, key: string) => {
+      headers[key] = val;
+    });
```

Comment:
style: Prefer for loop
```suggestion
    // convert Headers object into a plain JS object
    const headers: Record<string, string> = {};
    for (const [key, value] of response.headers.entries()) {
      headers[key] = val;
    }
```

Reply:
I wish, but it seems like it doesn't have entries :(
<img width="977" alt="image" src="https://github.com/winglang/wing/assets/39455181/e688218a-8bb5-4f4d-84c8-13136c34168e">